### ISSUE TITLE
lthread _switch() only save/restore general purpose registers

### ISSUE THREAD

**Issue by bodzhang** [lsds/sgx-lkl#747]
lthread [_switch()](https://github.com/lsds/sgx-lkl/blob/12fd34de68b88eefd16d27c9f07f7bcd89d89e65/src/sched/lthread.c#L103) does not save/restore FP, SSE or other CPU context. If the multiple lthreads running use FP, SSE or other advanced feature set instructions, the context switch between lthreads might corrupt the CPU context. 

It's known that many crypto implementations use SSE or other advanced feature set instructions for performance and security reason (for example, AES-NI based implementation to addresses certain side-channel attack risks).

**Comment by davidchisnall** [comment]
I don’t believe this is a bug. The lthread switch routine is a cooperative switch and so needs only to preserve callee-save registers. In the SysV psABI For x86-64, all floating-point and vector register are caller-save. The x87 CW register may need saving.

**Comment by davidchisnall** [comment]
The SSE2 CSW also probably needs saving / restoring.
